data_structures/linked_list/is_palindrome.py

```diff
@@ -1,4 +1,35 @@
+class ListNode:
```

Comment:
Please make this a dataclass like:
https://github.com/TheAlgorithms/Python/blob/895dffb412d80f29c65a062bf6d91fd2a70d8818/data_structures/linked_list/print_reverse.py#L4-L10

Reply:
Will do

Reply:
I updated code now, since ruff is complaining that i can't use keyword next. i update code reference from <code>next</code> to <code>next_node</code>

Reply:
Any pointers for pre-commit failure;
data_structures/linked_list/is_palindrome.py:7: error: Incompatible types in assignment (expression has type "None", variable has type "ListNode")  [assignment]?